zips/zip-0228.rst

```diff
@@ -254,62 +169,56 @@ We cover here the reasons for the different design choices in different sections
+As a result, it is plausible that the party performing the matching and sending it for settlement will not match orders having an ``nAGExpiryHeight`` that is near enough that it might not get settled on time due to the expected delay between the matching and the inclusion within a block.
 
-Rationale for Spend Authorization Signature Changes
-```````````````````````````````````````````````````
+Rationale for v1 Signature Digests and Spend Authorization Signature Changes
+````````````````````````````````````````````````````````````````````````````
 
-In the current (NU5) Zcash protocol, each Action includes a Spend Authorization Signature [#protocol-spendauthsig]_ that binds a specific spend instruction to a specific transaction and prevents replay attacks.
+In the OrchardZSA protocol, each Action includes a Spend Authorization Signature [#protocol-spendauthsig]_ that binds a specific spend instruction to a specific transaction and prevents replay attacks.
 However, in the context of Asset Swaps, there are two problems that prevent this mechanism from being carried over unchanged.
 
-- The SIGHASH represents a hash over a full consensus-compliant transaction object. Since a Swap Order is a trade intent to be used by the party performing the matching to form a valid bundle/transaction, the sender of the Swap Order cannot compute a SIGHASH as per ZIP 244. This trader doesn't know all the transaction fields in advance, as they are set when the full bundle transaction is formed and sent to the chain.
-- The ``timeLimit`` parameter is a new addition to the protocol, and is therefore not included in the existing SIGHASH. Swap Orders need to be made non-malleable in order to ensure that the party performing the matching cannot change the time limit unilaterally.
+- The SIGHASH represents a hash over a full consensus-compliant transaction object. Since a Swap Order is a trade intent to be used by the party performing the matching to form a valid bundle/transaction, the sender of the Swap Order cannot compute a v0 SIGHASH as per ZIP 246. This trader doesn't know all the transaction fields in advance, as they are set when the full bundle transaction is formed and sent to the chain.
 
 To make sure the time limit isn't malleable, the sender signs the information in the Action Group of the Order. 
-The ``orchard_action_groups_digest`` contains this information, which is why we use this field from the SIGHASH computation. 
-A party trying to tweak the time limit of received orders while matching orders will then need to be able to successfully forge the sender's Spend Authorization Signature.
+The ``orchard_action_groups_digest`` contains all the information that is in a Swap Order, which is why we use this for the v1 Signature Digest. 
+A party trying to tweak the expiry height or other details of received orders while matching orders will then need to be able to successfully forge the sender's Spend Authorization Signature.
```

Comment:
No need to forge..
Just "expiry height and other fields are locked by the sighash signature to prevent malleability.." or similar.